api/search.rb

```diff
@@ -125,14 +125,21 @@ def get_threads(context, group_ids, local_params, search_text)
   result_obj.to_json
 end
 
+error Sinatra::Param::InvalidParameterError do
+  # NOTE (CCB): The current behavior of the service is to return a seemingly positive response
+  # for an invalid request. In the future the API's contract should be modified so that HTTP 400
+  # is returned. This informs the client that the request was invalid, rather than having to guess
+  # about an empty response body.
+  [200, '{}']
+end
+
 get "#{APIPREFIX}/search/threads" do
+  param :text, String, required: true
+  param :context, String, default: 'course'
+  param :sort_order, String, in: %w(asc desc), transform: :downcase
```

Comment:
I found where `sort_key` is used (https://github.com/edx/cs_comments_service/pull/225/files#diff-c3ad3997b2a26ed201e7c27348b6ce2eR111), but it seems like `sort_order` is not. I'll pull out `sort_order` here and in tests.